### DISCUSSION TITLE
Dialogic 1.X Character game position

### DISCUSSION THREAD

**Discussion by PuteraSinuraya** [dialogic-godot/dialogic#1504] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/60869659/230545294-1f23d9b8-f9d9-4bdf-8d64-7f0d17ac6995.png)

From the picture above, there is a lot of empty space on the screen's left and right hand corner, and I want to take advantage of these empty spaces. However, I don't know how to modify the exact position that a sprite will occupy. At the moment all the 5 positions mainly occupy the center of screen.

Any help?

**Comment by PuteraSinuraya** [comment]
![image](https://user-images.githubusercontent.com/60869659/230561734-f7969f78-3f8a-4492-bbe2-b1e3efdcca23.png)
Oooh, this is the responsible code isn't it?
Now I want to increase the number of positions from 5 to 7, do you have any tips on how to do that?

**Reply by coppolaemilio** [reply]
I don't remember at the moment how the code it's set up, but I'll try to refresh my memory and let you know 😅

**Comment by zaknafean** [comment]
We're currently experimenting with some 'lost' code from another contributor that may help your situation. Can you try replacing that function with

```
func move_to_position(position_offset, time = 0.5):
    var reference = get_viewport().get_visible_rect().size
    var positions = {
        'left': Vector2(-2*(reference.x/7), 0),
        'right': Vector2(2*(reference.x/7), 0),
        'center': Vector2(0, 0),
        'center_right': Vector2(-(reference.x/7), 0),
        'center_left': Vector2(reference.x/7, 0)
        }
```

 And letting us know if it works for you?

**Comment by zaknafean** [comment]
This should now be merged into the Dialogic 1 branch.  You can grab it here since 1.5 still isn't ready for the asset store yet.
https://github.com/coppolaemilio/dialogic/tree/dialogic-1

**Comment by PuteraSinuraya** [comment]
Hang on, what do you mean when you said it has been merged into dialogic 1 branch? Did you mean dialogic has been updated to include the feature that I just requested?

I am using Dialogic in conjunction with Godot-Spine. Because of this, I also edited/modded my Dialogic 1.4.4 to suit my project needs (such as the json file extension problem that spine-godot has). I much prefer to modify my current dialogic if there is only small changes in the latest version of dialogic.

**Comment by zaknafean** [comment]
Ya, we updated Dialogic 1 branch to include the fix you requested.

But the code that was added is here. Then ya, replace the code you found above with this.
```
func move_to_position(position_offset, time = 0.5):
    var reference = get_viewport().get_visible_rect().size
    var positions = {
        'left': Vector2(-2*(reference.x/7), 0),
        'right': Vector2(2*(reference.x/7), 0),
        'center': Vector2(0, 0),
        'center_right': Vector2(reference.x/7, 0),
        'center_left': Vector2(-(reference.x/7), 0)
        }
```

**Reply by PuteraSinuraya** [reply]
I have tried the code. It runs..., but I was asking how to increase the number of position slots from 5 to 7.  Ideally, there would be 2 new position slots like "center_left2" and "center_right2".

**Reply by zaknafean** [reply, marked as answer]
Adding the slots is pretty easy technically, you could extend that code to do it well enough by following the pattern provided. Updating the UI to select them easily is a much more involved task however. Definitely out of scope for me tonight. I can look up the relevant scenes tomorrow to at least point you in the right direction

**Reply by PuteraSinuraya** [reply]
Thank you. Second thought I am quite happy with five slots.
